### PR TITLE
Update find, favorite, goto, undo, redo sections for UG

### PR DESCRIPTION
**Pull request by Na-Nazhou**
Can be merged after #148

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -173,35 +174,33 @@ Edits the name and folder of the 1st bookmark to be CS2103T Website and AY1920-S
...
-* Words will be partially matched e.g. `Video` will match `Videos`.
+* For keywords, words will be partially matched e.g. `Video` will match `Videos`.
+* For tags and folders, words will be fully matched.
 * Bookmarks matching at least one search criterion will be returned (i.e. OR search).
 ****
 
 Examples:
 
-* `*find* Website Video t/Fun will` +
+* `*find* Website Video will t/Fun` +
```
Comment (openorclose) [inline docs/UserGuide.adoc]: actually what's the will for?

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -264,30 +260,69 @@ Examples:
...
 `reminder edit`, `reminder delete`, etc).
+* The step must be a positive integer `1`, `2`, `3`, …​
 ****
 
-=== Redoing the previously undone command: *`redo`*
+Examples:
 
-Reverses the most recent `*undo*` command.
+* `*undo*` +
+Restores the list to the state before the previous undoable command was	executed.
```
Comment (TSAI-HSIAO-HAN) [inline docs/UserGuide.adoc]:
```suggestion
Restores the list to the state before the previous undoable command was executed.
```

File: docs/UserGuide.adoc (hunk 1 above)
Comment (Na-Nazhou) [inline docs/UserGuide.adoc]:
```suggestion
* `*find* Website Video t/Fun` +
```

File: docs/UserGuide.adoc [hunk 3]
```diff
@@ -173,35 +174,33 @@ Edits the name and folder of the 1st bookmark to be CS2103T Website and AY1920-S
...
+* For keywords, words will be partially matched e.g. `Video` will match `Videos`.
+* For tags and folders, words will be fully matched.
 * Bookmarks matching at least one search criterion will be returned (i.e. OR search).
 ****
 
 Examples:
 
-* `*find* Website Video t/Fun will` +
+* `*find* Website Video will t/Fun` +
 Returns CS2103T Website, Video Streaming, and all bookmarks tagged with Fun.
```
Comment (Na-Nazhou) [inline docs/UserGuide.adoc]:
```suggestion
Returns all the bookmarks with CS2103T or Video in its name or URL, and all bookmarks tagged with Fun.
```

**Pull request merged**